### PR TITLE
Add EIP: Reduced Attestation Format

### PR DESCRIPTION
**Pull request by Giulio2002**
**ATTENTION: ERC-RELATED PULL REQUESTS NOW OCCUR IN [ETHEREUM/ERCS](https://github.com/ethereum/ercs)**

--

When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-7610.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+---
+eip: 7610
```
Comment (abcoathup) [inline EIPS/eip-7610.md:2]:
```suggestion
eip: 7670
```
Assigning next sequential EIP/ERC/RIP number.
Numbering changed to sequential from 7500.

Please also update the filename.

File: EIPS/eip-7610.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+title: Reduced Attestation Format
+description: <Description is one full (short) sentence>
+author: Giulio Rebuffo (@Giulio2002)
+discussions-to: https://ethereum-magicians.org/t/reduced-attestation-format-for-gossiping-attestations/19458
```
Comment (abcoathup) [inline EIPS/eip-7610.md:6]:
```suggestion
discussions-to: https://ethereum-magicians.org/t/eip-7670-reduced-attestation-format-for-gossiping-attestations/19458
```

File: EIPS/eip-7610.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-03-31
+---
+
+## Abstract
+
+This EIP introduces a reduced version of the `Attestation`sformat to reduce the size of attestations transmitted on the Consensus Layer's gossip network. This is achieved by replacing Checkpoint objects in gossiped `AttestationData`s with their respective 8-byte digests.
```
Comment (g11tech) [inline EIPS/eip-7610.md:15]:
```suggestion
This EIP introduces a reduced version of the `Attestation` format to reduce the size of attestations transmitted on the Consensus Layer's gossip network. This is achieved by replacing Checkpoint objects in gossiped `AttestationData` with their respective 8-byte digests.
```
imo looks cleaner this way

File: EIPS/eip-7610.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+## Motivation
+
+Attestations currently dominate the bandwidth usage of the Consensus Layer's gossip network (approx. 90% of the traffic). By optimizing the size of `AttestationData`, the aim is to reduce network load to improve node performance, and, most importantly, allow for greater throughput of other significant consensus data, such as Blobs. As a matter of fact, the current `Attestation` structure is not optimized for network transmission, leading to unnecessarily high bandwidth consumption. As a matter of fact, the application of Snappy block compression only reduces the size of an attestation by less than 10%.
```
Comment (g11tech) [inline EIPS/eip-7610.md:19]:
```suggestion
Attestations currently dominate the bandwidth usage of the Consensus Layer's gossip network (approx. 90% of the traffic). By optimizing the size of `AttestationData`, the aim is to reduce network load to improve node performance, and, most importantly, allow for greater throughput of other significant consensus data, such as Blobs. As a matter of fact, the current `Attestation` structure is not optimized for network transmission, leading to high bandwidth consumption. As a matter of fact, the application of Snappy block compression only reduces the size of an attestation by less than 10%.
```
+ may be a diagram with subnets subscribed with traffic a node experiences will provide a very nice context

File: EIPS/eip-7610.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    aggregation_bits: Bitlist[MAX_VALIDATORS_PER_COMMITTEE]
+    data: ReducedAttestationData
+    signature: BLSSignature
+```
+
+```python
+class ReducedAttestationData(Container):
+    class AttestationData(Container):
+    slot: Slot
+    index: CommitteeIndex
```
Comment (g11tech) [inline EIPS/eip-7610.md:56]: index is being pulled out of attestation data: https://eips.ethereum.org/EIPS/eip-7549

Comment (g11tech) [inline EIPS/eip-7610.md:56]: index is being pulled out of attestation data: https://eips.ethereum.org/EIPS/eip-7549

File: EIPS/eip-7610.md (hunk 4 above)
Comment (Giulio2002) [reply EIPS/eip-7610.md:19]: I do not actually have that for now, will work on it this week.

Comment (g11tech) [reply EIPS/eip-7610.md:19]: ok let me know if you want to add that or you think this EIP is ready for draft?

File: EIPS/eip-7670.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
+---
+eip: 7670
+title: Reduced Attestation Format
+description: <Description is one full (short) sentence>
```
Comment (SamWilsn) [inline EIPS/eip-7670.md:4]:
```suggestion
description: Introduces a shorter format for attestations transmitted on the gossip network.
```

File: EIPS/eip-7670.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+created: 2024-03-31
+---
+
+## Abstract
+
+This EIP introduces a reduced version of the `Attestation` format to reduce the size of attestations transmitted on the Consensus Layer's gossip network. This is achieved by replacing Checkpoint objects in gossiped `AttestationData` with their respective 8-byte digests.
+
+## Motivation
+
+Attestations currently dominate the bandwidth usage of the Consensus Layer's gossip network (approx. 90% of the traffic). By optimizing the size of `AttestationData`, the aim is to reduce network load to improve node performance, and, most importantly, allow for greater throughput of other significant consensus data, such as Blobs. As a matter of fact, the current `Attestation` structure is not optimized for network transmission, leading to unnecessarily high bandwidth consumption. As a matter of fact, the application of Snappy block compression only reduces the size of an attestation by less than 10%.
```
Comment (SamWilsn) [inline EIPS/eip-7670.md:19]:
```suggestion
Attestations currently dominate the bandwidth usage of the Consensus Layer's gossip network (approx. 90% of the traffic). By optimizing the size of `AttestationData`, the aim is to reduce network load to improve node performance, and, most importantly, allow for greater throughput of other significant consensus data, such as Blobs. As a matter of fact, the current `Attestation` structure is not optimized for network transmission, leading to unnecessarily high bandwidth consumption. Furthermore, the application of Snappy block compression only reduces the size of an attestation by less than 10%.
```

File: EIPS/eip-7670.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+## Rationale
+
+The decision of introducing a cache stems from the fact that many consensus clients already maintain caches of checkpoints, implying that adapting to a system where checkpoints are referenced through digests would not significantly increase the protocol's complexity.
+
+The choice to use 8 or 16 bytes for `CHECKPOINT_DIGEST_LENGTH` balances data efficiency with security. Considering Ethereum's slot timing, attackers have limited time to generate collisions. This constraint makes shorter digests practical, as the risk of collision within the short slot intervals is minimal. Hence, using the first few bytes of the hash as digests is efficient for network propagation without significantly increasing collision risk.
+
+## Backwards Compatibility
+
+Backward incompatible.
```
Comment (SamWilsn) [inline EIPS/eip-7670.md:90]: You should describe the backwards incompatibility. Like "clients using a pre-7670 network format will be unable to communicate with clients post-7670."

**Pull request closed without merging**